PHP / PHP Coding Standards Fixer (pull_request)  -> there's no more explaination about error ? I have juste Error: Process completed with exit code 8

Do you have any idea ?

--- a/classes/ProductAssembler.php
+++ b/classes/ProductAssembler.php
@@ -50,11 +50,15 @@ class ProductAssemblerCore
      * @param array $rawProduct
      *
      * @return array
+     *
+     * @throws PrestaShopDatabaseException
      */
-    private function addMissingProductFields(array $rawProduct)
+    private function addMissingProductFields(array $rawProduct): array
     {
-        $idShop = (int) $this->searchContext->getIdShop();
-        $idLang = (int) $this->searchContext->getIdLang();
+        $idShop = $this->searchContext->getIdShop();
+        $idShopGroup = $this->searchContext->getIdShopGroup();
+        $isStockSharingBetweenShopGroupEnabled = $this->searchContext->isStockSharingBetweenShopGroupEnabled();
+        $idLang = $this->searchContext->getIdLang();
         $idProduct = (int) $rawProduct['id_product'];
         $prefix = _DB_PREFIX_;
 
@@ -83,10 +87,17 @@ class ProductAssemblerCore
                     ON pl.id_product = p.id_product
                     AND pl.id_shop = $idShop
                     AND pl.id_lang = $idLang
-                LEFT JOIN {$prefix}stock_available sa
-			        ON sa.id_product = p.id_product
-			        AND sa.id_shop = $idShop
-                LEFT JOIN {$prefix}product_shop ps
+                LEFT JOIN {$prefix}stock_available sa ";
+
+        if ($isStockSharingBetweenShopGroupEnabled) {
+            $sql .= "  ON sa.id_product = p.id_product
+			        AND sa.id_shop = 0
+			        AND sa.id_shop_group = $idShopGroup ";
+        } else {
+            $sql .= "  ON sa.id_product = p.id_product
+			        AND sa.id_shop = $idShop ";
+        }
+        $sql .= "LEFT JOIN {$prefix}product_shop ps
 			        ON ps.id_product = p.id_product
 			        AND ps.id_shop = $idShop
 			    WHERE p.id_product = $idProduct
@@ -106,6 +117,8 @@ class ProductAssemblerCore
      * @param array $rawProduct
      *
      * @return mixed
+     *
+     * @throws PrestaShopDatabaseException
      */
     public function assembleProduct(array $rawProduct)
     {

--- a/src/Core/Product/Search/ProductSearchContext.php
+++ b/src/Core/Product/Search/ProductSearchContext.php
@@ -50,10 +50,24 @@ class ProductSearchContext
      */
     private $idCustomer;
 
+    /**
+     * @var int the Shop Group id
+     */
+    private $idShopGroup;
+
+    /**
+     * @var bool if the sharing stock is enable
+     */
+    private $stockSharingBetweenShopGroupEnabled = false;
+
     public function __construct(Context $context = null)
     {
         if ($context) {
+            $shopGroup = $context->shop->getGroup();
+
             $this->idShop = $context->shop->id;
+            $this->idShopGroup = $shopGroup->id;
+            $this->stockSharingBetweenShopGroupEnabled = (bool) $shopGroup->share_stock;
             $this->idLang = $context->language->id;
             $this->idCurrency = $context->currency->id;
             $this->idCustomer = $context->customer->id;
@@ -63,7 +77,7 @@ class ProductSearchContext
     /**
      * @param int $idShop
      *
-     * @return $this
+     * @return self
      */
     public function setIdShop($idShop)
     {
@@ -83,7 +97,7 @@ class ProductSearchContext
     /**
      * @param int $idLang
      *
-     * @return $this
+     * @return self
      */
     public function setIdLang($idLang)
     {
@@ -103,7 +117,7 @@ class ProductSearchContext
     /**
      * @param int $idCurrency
      *
-     * @return $this
+     * @return self
      */
     public function setIdCurrency($idCurrency)
     {
@@ -123,7 +137,7 @@ class ProductSearchContext
     /**
      * @param int $idCustomer
      *
-     * @return $this
+     * @return self
      */
     public function setIdCustomer($idCustomer)
     {
@@ -138,5 +152,45 @@ class ProductSearchContext
     public function getIdCustomer()
     {
         return $this->idCustomer;
+    }
+
+    /**
+     * @return int the Shop Group Iid
+     */
+    public function getIdShopGroup(): int
+    {
+        return $this->idShopGroup;
+    }
+
+    /**
+     * @param int $idShopGroup
+     *
+     * @return self
+     */
+    public function setIdShopGroup(int $idShopGroup): self
+    {
+        $this->idShopGroup = $idShopGroup;
+
+        return $this;
+    }
+
+    /**
+     * @return bool if sharing stock is enable
+     */
+    public function isStockSharingBetweenShopGroupEnabled(): bool
+    {
+        return $this->stockSharingBetweenShopGroupEnabled;
+    }
+
+    /**
+     * @param bool $stockSharingBetweenShopGroupEnabled
+     *
+     * @return self
+     */
+    public function setStockSharingBetweenShopGroupEnabled(bool $stockSharingBetweenShopGroupEnabled): self
+    {
+        $this->stockSharingBetweenShopGroupEnabled = $stockSharingBetweenShopGroupEnabled;
+
+        return $this;
     }
 }

